napalm_junos/junos.py

```diff
@@ -854,20 +854,38 @@ def get_interfaces_ip(self):
 
     def get_mac_address_table(self):
 
-        mac_address_table = list()
+        mac_address_table = []
 
         mac_table = junos_views.junos_mac_address_table(self.device)
+
+        if self.device.facts.get('personality', '') in ['SWITCH']:  # for EX & QFX devices
```

Comment:
Just name it differently, for example, `junos_mac_address_table_ex_qfx` or something meaningful and then:

```
if self.device.facts.get('personality', '') in ['SWITCH']:  # for EX & QFX devices
     mac_table = junos_views.junos_mac_address_table_ex_qfx(self.device)
else:
     mac_table = junos_views.junos_mac_address_table(self.device)
```

Wouldn't that work?